packages/teleterm/src/ui/Navigator/ExpanderConnections/useExpanderConnections.ts

```diff
@@ -18,81 +18,108 @@ import * as Icons from 'design/Icon';
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import AppContext from 'teleterm/ui/appContext';
 import * as types from 'teleterm/ui/Navigator/types';
+import { Document, DocumentTshNode } from 'teleterm/ui/services/docs/types';
+import React, { useEffect } from 'react';
 
 export default function useExpanderConnections() {
   const ctx = useAppContext();
-  const items = initItems(ctx);
+  const items = getWorkspaceItems(ctx);
+  const [, rerender] = React.useState<any>();
 
   // subscribe to services
   ctx.clustersService.useState();
   ctx.docsService.useState();
 
+
+  //TODO: think how to use useStore for services from context bridge
+  useEffect(() => {
+    function onChange() {
+      rerender({});
+    }
+
+    ctx.workspaceService.subscribe(onChange);
```

Comment:
curious if it needs to subscribe to the workspaceService? Would it be enough to use the service only for initial state?

Reply:
curious if it needs to subscribe to the workspaceService? Would it be enough to use the service only for initial state?